Toby dev psqlodbc tests bpchar

HackonToby opened the pull request:
### Description

[Describe what this change achieves]
 
### Issues Resolved

[List any issues this PR will resolve]


### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

faviansamatha committed:
Added initial psqlodbc datatype test for bigint

Signed-off-by: Favian (Ian) Samatha <ians@bitquilltech.com>

faviansamatha committed:
Created table in master_dbo schema

Signed-off-by: Favian (Ian) Samatha <ians@bitquilltech.com>

faviansamatha committed:
Refactored the tests, added more test cases, and split them into multiple tests

Signed-off-by: Favian (Ian) Samatha <ians@bitquilltech.com>

faviansamatha committed:
Initial commit for varchar tests

Signed-off-by: Favian (Ian) Samatha <ians@bitquilltech.com>

faviansamatha committed:
Addressed PR comments

Signed-off-by: Favian (Ian) Samatha <ians@bitquilltech.com>

faviansamatha committed:
Added arithmetic and view creation tests

Signed-off-by: Favian (Ian) Samatha <ians@bitquilltech.com>

faviansamatha committed:
Added initial varchar tests. Still need to clean up

Signed-off-by: Favian (Ian) Samatha <ians@bitquilltech.com>

faviansamatha committed:
init commit

Signed-off-by: Favian (Ian) Samatha <ians@bitquilltech.com>

faviansamatha committed:
added table_creat_fail test

Signed-off-by: Favian (Ian) Samatha <ians@bitquilltech.com>

faviansamatha committed:
deleted bigint

Signed-off-by: Favian (Ian) Samatha <ians@bitquilltech.com>

faviansamatha committed:
saving changes before rebase

Signed-off-by: Favian (Ian) Samatha <ians@bitquilltech.com>

faviansamatha committed:
Added insertion_success

Signed-off-by: Favian (Ian) Samatha <ians@bitquilltech.com>

faviansamatha committed:
Added table creation fail test

Signed-off-by: Favian (Ian) Samatha <ians@bitquilltech.com>

faviansamatha committed:
Cleaned up for first revision

Signed-off-by: Favian (Ian) Samatha <ians@bitquilltech.com>

faviansamatha committed:
updated UPDATE_FAIL test to close statement handle

Signed-off-by: Favian (Ian) Samatha <ians@bitquilltech.com>

faviansamatha committed:
Deleted varchar file

Signed-off-by: Favian (Ian) Samatha <ians@bitquilltech.com>

Commit:
Float test

Commit:
bpchar test

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+  DATATYPE + "(" + std::to_string(COL_LENGTH[3]) + ")"
+};
+
+vector<pair<string, string>> TABLE_COLUMNS_NVARCHAR = {
```
```suggestion
const vector<pair<string, string>> TABLE_COLUMNS_NVARCHAR = {
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+const string STRING_1 = "a";
+const string STRING_20 = "0123456789abcdefghij";
+
+class PSQL_DataTypes_bpchar : public testing::Test{
+
+  void SetUp() override {
+
+    OdbcHandler test_setup;
+    test_setup.ConnectAndExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+  }
+
+  void TearDown() override {
+
+    OdbcHandler test_cleanup;
```
```suggestion
class PSQL_DataTypes_bpchar : public testing::Test{
  void SetUp() override {
    OdbcHandler test_setup;
    test_setup.ConnectAndExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
  }

  void TearDown() override {
    OdbcHandler test_cleanup;
```

ColinKYuen replied:
After reading thru, this function isn't needed. Using `string(LENGTH, ' ')`

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+  return ret;
+};
+// helper function to initialize insert string (1, "", "", ""), etc.
+string InitializeInsertString(const vector<vector<string>> &inserted_values) {
```
Since this function seems to be used quite a bit, might make sense to just move it into `query_generator.h/cpp`

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+  }
+  return ret;
+};
+// helper function to initialize insert string (1, "", "", ""), etc.
```
```suggestion
string operator_multiple(const string& s, unsigned int n){
  string ret{};
  for (unsigned int i = 0; i < n; i++){
    ret += s;
  }
  return ret;
};

// helper function to initialize insert string (1, "", "", ""), etc.
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+  const int BUFFER_SIZE = 256;
+  char name[BUFFER_SIZE];
+  char suffix[BUFFER_SIZE];
+  char prefix[BUFFER_SIZE];
```
```suggestion
```
Unused

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+
+TEST_F(PSQL_DataTypes_bpchar, ColAttributes) {
+
+  const int LENGTH_EXPECTED = 10;
```
```suggestion
TEST_F(PSQL_DataTypes_bpchar, ColAttributes) {
  const int LENGTH_EXPECTED = 10;
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+  const string SUFFIX_EXPECTED = "'";
+  const int BYTES_EXPECTED = 2;
+
+  const int BUFFER_SIZE = 256;
```
Can be made global

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+
+TEST_F(PSQL_DataTypes_bpchar, Table_Create_Fail) {
+
+  vector<vector<pair<string, string>>> invalid_columns{
```
```suggestion
TEST_F(PSQL_DataTypes_bpchar, Table_Create_Fail) {
  vector<vector<pair<string, string>>> invalid_columns{
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+
+TEST_F(PSQL_DataTypes_bpchar, Insertion_Success) {
+
+  const int BUFFER_LENGTH = 8192;
```
Can be made global & probably half this size

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+
+  // initialize bind_columns
+  for (int i = 0; i < NUM_COLS; i++) {
+    tuple<int, int, SQLPOINTER, int, SQLLEN*> tuple_to_insert(i+1, SQL_C_CHAR, (SQLPOINTER) &col_results[i], BUFFER_LENGTH, &col_len[i]);
```
```suggestion
    tuple<int, int, SQLPOINTER, int, SQLLEN*> tuple_to_insert(i + 1, SQL_C_CHAR, (SQLPOINTER) &col_results[i], BUFFER_LENGTH, &col_len[i]);
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+  // Make sure inserted values are correct
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(bind_columns));
+
+  for (int i = 0; i < inserted_values.size(); ++i) {
+    
+    rcode = SQLFetch(odbcHandler.GetStatementHandle()); // retrieve row-by-row
```
```suggestion
  for (int i = 0; i < inserted_values.size(); ++i) {    
    rcode = SQLFetch(odbcHandler.GetStatementHandle()); // retrieve row-by-row
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+    rcode = SQLFetch(odbcHandler.GetStatementHandle()); // retrieve row-by-row
+    ASSERT_EQ(rcode, SQL_SUCCESS);
+
+    for (int j = 0; j < NUM_COLS; j++) {
+      std::cout<<"Current string is: "<<inserted_values[i][j]<<'\n'<<"Current number of space: "<<COL_LENGTH[j]<<" ==>> "<<inserted_values[i][j].size()<<"\n";
+      if (inserted_values[i][j] != "NULL") {
```
```suggestion
    for (int j = 0; j < NUM_COLS; j++) {
      if (inserted_values[i][j] != "NULL") {
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+        ASSERT_EQ(string(col_results[j]), inserted_values[i][j]+ operator_multiple(" ",(COL_LENGTH[j]-inserted_values[i][j].size())));
+        ASSERT_EQ(col_len[j], COL_LENGTH[j]);
+      } 
+      // else if(inserted_values[i][j]==""){
+      //   ASSERT_EQ(string(col_results[j]), inserted_values[i][j]+(std::string ((COL_LENGTH[j]-inserted_values[i][j].size()," "))));
+      //   ASSERT_EQ(col_len[j], COL_LENGTH[j]);
+      // }
```
Remove if not needed

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+      //   ASSERT_EQ(string(col_results[j]), inserted_values[i][j]+(std::string ((COL_LENGTH[j]-inserted_values[i][j].size()," "))));
+      //   ASSERT_EQ(col_len[j], COL_LENGTH[j]);
+      // }
+      else 
+        ASSERT_EQ(col_len[j], SQL_NULL_DATA);
+    }
```
```suggestion
      else {
        ASSERT_EQ(col_len[j], SQL_NULL_DATA);
      }
    }
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+    for (int j = 0; j < NUM_COLS; j++) {
+      std::cout<<"Current string is: "<<inserted_values[i][j]<<'\n'<<"Current number of space: "<<COL_LENGTH[j]<<" ==>> "<<inserted_values[i][j].size()<<"\n";
+      if (inserted_values[i][j] != "NULL") {
+        ASSERT_EQ(string(col_results[j]), inserted_values[i][j]+ operator_multiple(" ",(COL_LENGTH[j]-inserted_values[i][j].size())));
```
```suggestion
        ASSERT_EQ(string(col_results[j]), inserted_values[i][j]+ std::string(COL_LENGTH[j] - inserted_values[i][j].size(), ' '));
```
Might need to double check the brackets

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+  odbcHandler.CloseStmt();
+
+  // Insert invalid values in table and assert error
+  for (int i = 0; i < inserted_values.size(); i++) {
+
+    string insert_string = "(";
```
```suggestion
  for (int i = 0; i < inserted_values.size(); i++) {
    string insert_string = "(";
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+
+  // initialize bind_columns
+  for (int i = 0; i < NUM_COLS; i++) {
+    tuple<int, int, SQLPOINTER, int, SQLLEN*> tuple_to_insert(i+1, SQL_C_CHAR, (SQLPOINTER) &col_results[i], BUFFER_LENGTH, &col_len[i]);
```
```suggestion
    tuple<int, int, SQLPOINTER, int, SQLLEN*> tuple_to_insert(i + 1, SQL_C_CHAR, (SQLPOINTER) &col_results[i], BUFFER_LENGTH, &col_len[i]);
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+    ASSERT_EQ(rcode, SQL_SUCCESS);
+
+    for (int j = 0; j < NUM_COLS; j++) {
+      ASSERT_EQ(string(col_results[j]), inserted_values[i][j]+ operator_multiple(" ",(COL_LENGTH[j]-inserted_values[i][j].size())));
```
```suggestion
      ASSERT_EQ(string(col_results[j]), inserted_values[i][j]+ string(COL_LENGTH[j] - inserted_values[i][j].size(), ' '));
```
Double check brackets again

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+  odbcHandler.CloseStmt();
+
+  for (int i = 0; i < updated_values.size(); i++) {
+
+    vector<pair<string,string>> update_col;
```
```suggestion
  for (int i = 0; i < updated_values.size(); i++) {
    vector<pair<string,string>> update_col;
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+    vector<pair<string,string>> update_col;
+    // setup update column
+    for (int j = 0; j <NUM_COLS; j++) {
+      string value = string("'") + updated_values[i][j] + string("'");
```
```suggestion
      string value = "'" + updated_values[i][j] + "'";
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+    odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {COL_NAMES[0]}));
+    rcode = SQLFetch(odbcHandler.GetStatementHandle());
+
+    for (int j = 0; j < NUM_COLS; j++) {
+      
+      ASSERT_EQ(string(col_results[j]), updated_values[i][j]+ operator_multiple(" ",(COL_LENGTH[j]-updated_values[i][j].size())));
```
```suggestion
    for (int j = 0; j < NUM_COLS; j++) {
      ASSERT_EQ(string(col_results[j]), inserted_values[i][j]+ string(COL_LENGTH[j] - inserted_values[i][j].size(), ' '));
```
Double check brackets again

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+const string DATATYPE = "sys.bpchar";
+const int NUM_COLS = 4;
+const string COL_NAMES[NUM_COLS] = {"pk", "nvarchar_1", "nvarchar_4000", "nvarchar_20"};
+const int COL_LENGTH[NUM_COLS] = {10, 1, 4000, 20};
```
Clears up when you do 
`ASSERT_EQ(col_length, COL_LENGTH)`
```suggestion
const int COL_LENGTH_EXPECTED[NUM_COLS] = {10, 1, 4000, 20};
```
Elsewhere too

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_bpchar, Update_Success) {
+
+  const int BUFFER_LENGTH = 8192;
```
```suggestion
TEST_F(PSQL_DataTypes_bpchar, Update_Success) {
  const int BUFFER_LENGTH = 8192;
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_bpchar, Update_Fail) {
+
+  const int BUFFER_LENGTH = 8192;
```
```suggestion
TEST_F(PSQL_DataTypes_bpchar, Update_Fail) {
  const int BUFFER_LENGTH = 8192;
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+
+  // initialize bind_columns
+  for (int i = 0; i < NUM_COLS; i++) {
+    tuple<int, int, SQLPOINTER, int, SQLLEN*> tuple_to_insert(i+1, SQL_C_CHAR, (SQLPOINTER) &col_results[i], BUFFER_LENGTH, &col_len[i]);
```
```suggestion
    tuple<int, int, SQLPOINTER, int, SQLLEN*> tuple_to_insert(i + 1, SQL_C_CHAR, (SQLPOINTER) &col_results[i], BUFFER_LENGTH, &col_len[i]);
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+    ASSERT_EQ(rcode, SQL_SUCCESS);
+
+    for (int j = 0; j < NUM_COLS; j++) {
+      ASSERT_EQ(string(col_results[j]), inserted_values[i][j]+ operator_multiple(" ",(COL_LENGTH[j]-inserted_values[i][j].size())));
```
```suggestion
      ASSERT_EQ(string(col_results[j]), inserted_values[i][j]+ string(COL_LENGTH[j] - inserted_values[i][j].size(), ' '));
```
Double check brackets

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+
+  // Assert that the results did not change
+  for (int i = 0; i < NUM_COLS; i++) {
+    ASSERT_EQ(string(col_results[i]), inserted_values[0][i]+ operator_multiple(" ",(COL_LENGTH[i]-inserted_values[0][i].size())));
```
```suggestion
    ASSERT_EQ(string(col_results[i]), inserted_values[0][i] + std::string(COL_LENGTH[j] - inserted_values[0][j].size(), ' '));
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_bpchar, View_creation) {
+
+  const string VIEW_QUERY = "SELECT * FROM " + TABLE_NAME;
```
```suggestion
TEST_F(PSQL_DataTypes_bpchar, View_creation) {
  const string VIEW_QUERY = "SELECT * FROM " + TABLE_NAME;
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+
+  // initialize bind_columns
+  for (int i = 0; i < NUM_COLS; i++) {
+    tuple<int, int, SQLPOINTER, int, SQLLEN*> tuple_to_insert(i+1, SQL_C_CHAR, (SQLPOINTER) &col_results[i], BUFFER_LENGTH, &col_len[i]);
```
```suggestion
    tuple<int, int, SQLPOINTER, int, SQLLEN*> tuple_to_insert(i + 1, SQL_C_CHAR, (SQLPOINTER) &col_results[i], BUFFER_LENGTH, &col_len[i]);
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+  // Make sure inserted values are correct
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(bind_columns));
+
+  for (int i = 0; i < inserted_values.size(); ++i) {
+    
+    rcode = SQLFetch(odbcHandler.GetStatementHandle()); // retrieve row-by-row
+    ASSERT_EQ(rcode, SQL_SUCCESS);
+
+    for (int j = 0; j < NUM_COLS; j++) {
+      
+      if (inserted_values[i][j] != "NULL") {
+
+        ASSERT_EQ(string(col_results[j]), inserted_values[i][j]+ operator_multiple(" ",(COL_LENGTH[j]-inserted_values[i][j].size())));
```
```suggestion
  for (int i = 0; i < inserted_values.size(); ++i) {    
    rcode = SQLFetch(odbcHandler.GetStatementHandle()); // retrieve row-by-row
    ASSERT_EQ(rcode, SQL_SUCCESS);

    for (int j = 0; j < NUM_COLS; j++) {      
      if (inserted_values[i][j] != "NULL") {
        ASSERT_EQ(string(col_results[j]), inserted_values[i][j] + std::string(COL_LENGTH[j] - inserted_values[i][j].size(), ' '));
```

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("VIEW", VIEW_NAME));
+}
```
New at end of file

ColinKYuen commented on test/odbc/test_pg_datatype_bpchar.cpp:
```diff
@@ -0,0 +1,650 @@
+#include <gtest/gtest.h>
```
Additional tests to add
- Unique and Composite Key

jusjosj committed:
Expand explain for PRINT, THROW, USE and RAISERROR (#561)

This commit extends the functionality of babelfish_showplan_all to include print, throw, raiserror, and use
This is largely done in a manner consistent with the other explain functions existing throughout the codebase.

Task: BABEL-2902
Signed-off-by: Justin Jossick <jusjosj@amazon.com>

shah-nirmit committed:
Change github actions to point to correct engine dev branch (#649)

Authored-by: Nirmit Shah <nirmisha@amazon.com>
Signed-off-by: Kuntal Ghosh <kuntalgh@amazon.com>

Sairakan committed:
Fix table hint parsing (#637)

Previously, Babelfish would sometimes incorrectly parse a query that used a table alias with column definitions as a table hint and raise an error. This commit fixes the issue by having the parser only accept a specific defined list of keywords as possible table hints.

N.B. Babelfish will report incorrect table hint syntax as generic syntax errors instead of specifically table hint errors as SQL Server does. This commit does not attempt to mimic this behavior, as that would require a more extensive modification to the antlr parser (most likely requiring a new grammar rule to be added).

Task: BABEL-2805
Signed-off-by: Jason Teng jasonten@amazon.com

AmithKamath committed:
Add JDBC tests for CONVERT()

An issue BABEL-2054 was reported where CONVERT(datetime, string) failed
intermittently. However, this issue is not reproducible and this commit
adds JDBC tests to validate the scenario mentioned in the JIRA

Task: BABEL-2054

Signed-off-by: Amith Kamath <amithkam@amazon.com>

wboettge committed:
Login who owns database should always be dbo also without sysadmin role

In SQL Server, the login that creates a database is automatically the
database owner. This means that the login can always access the
databases they create.

However, previously in babelfish, if a login is stripped of the sysadmin
role after creating a database, and the login does not have a
user associated with the database, then the login can no longer access
the database. Now, when a login attempts to access and there
is not an associated user, the login will be considered the dbo and
allowed access.

Task: BABEL-3548

Signed-off-by: Walt Boettge <wboettge@amazon.com>

shah-nirmit committed:
Refactor Encoding Conversion Related Functions to improve Performance (#632)

Previously, we were using pg_any_to_server() API on receiving side for varchar, char and text datatypes.
pg_any_to_server() internally do the sys cache lookup which was expensive in some cases(varchar,char and text datatypes) and we really do not need to do it as TDS only supports collations which requires only few set of encodings. This commit adds new function encoding_conv_util() which would take src_encoding and dest_encoding as argument and will call appropriate encoding conversion functionality based on those arguments. This commit also copies few encoding conversion related functionality (TsqlLocalToUtf)from engine to common extension and extends them to return byte length of encoded string. 
 
Changes: 
1.  Remove duplicate encoding logic from babelfishpg_tsql extension
2. Introduce conditional handling for encoding conversion for Varchar, Char and Text datatypes on receiver side
3. Copy and enhance TsqlLocalToUtf from engine side to be used by various encoding functions


### Testing
- Collation TestFramework
  - Tested Collations
    - sql_latin1_general_cp1_cs_as
    - arabic_cs_as
    - greek_cs_as
    - turkish_cs_as
    - polish_cs_as
    - vietnamese_cs_as
    - ukrainian_cs_as
    - japanese_cs_as
    - korean_wansung_cs_as
- Manual testing with custom C# program

Task: BABEL-3542
Signed-off-by: Nirmit Shah <nirmisha@amazon.com>

ZhenyeLee committed:
Fix syscoumns test failure (#660)

The helper function in prepare file searching object_name by oid in both pg_class and pg_proc. However, pg_class and pg_proc may have the same oid and it cause test failure. In this PR, we break down the search into to functions, one is in pg_class, and another one is in pg_proc. And also removed order by id from verification file, because oid can be changed after MVU.

Signed-off-by: Zhenye Li zhenye@amazon.com

Commit:
Merge branch 'babelfish-for-postgresql:BABEL_2_X_DEV' into BABEL_2_X_DEV

Commit:
rebase,operators,keys added

Commit:
PR comments addressed, add comparison operators, one error in string_operators fixed

ColinKYuen commented:
PR made & Merged into upstream
